modules/core/src/main/java/org/apache/ignite/internal/commandline/CommonArgParser.java

```diff
@@ -336,4 +369,70 @@ private String securityWarningMessage(String password) {
 
         return String.format(pwdArgWarnFmt, password, password);
     }
+
+    /**
+     * Extracts user properties from property string or file.
+     *
+     * @param userPropStr {@code String} Property string.
+     * @param userPropPath {@code String} Property file path.
+     */
+    private Map<String, String> userProperties(String userPropStr, String userPropPath){
+        Map<String, String> res = parsePropertiesFromString(userPropStr);
+
+        parsePropertiesFromFile(userPropPath).forEach(res::putIfAbsent);
+
+        return res;
+    }
+
+    /**
+     * Extracts user properties from attribute string.
+     *
+     * @param propStr {@code String} Attribute string.
+     */
+    private Map<String, String> parsePropertiesFromString(String propStr) {
+        Map<String, String> res = new HashMap<>();
+
+        if(propStr == null)
+            return res;
+
+        final int partsOfPropStr = 2;
+
+        for (String prop : propStr.split(",")) {
+            if (!prop.contains("="))
+                throw new RuntimeException(String.format("Failed to parse property %s", prop));
```

Comment:
Is it common to use `RuntimeException` in control utility code to handle input exceptions? Applicable to `parsePropertiesFromFile` method as well. What will user see in case of such input exceptions? It might be useful to write tests checking it.

Reply:
Yes, we are already throwing RuntimeException in case of unexpected property.

The user will see error message like this:

Control utility [ver. 2.9.0-SNAPSHOT#20200327-sha1:1ae99695]
2020 Copyright(C) Apache Software Foundation
User: olegostanin
Time: 2020-03-30T09:54:56.505
Warning: --keystore-password is insecure. Whenever possible, use interactive prompt for password (just discard --keystore-password option).
Warning: --truststore-password is insecure. Whenever possible, use interactive prompt for password (just discard --truststore-password option).
Failed to read property file: /path_to_the_file
java.io.FileNotFoundException: /path_to_the_file (No such file or directory)
Command [] finished with code: 4
Control utility has completed execution at: 2020-03-30T09:54:56.522
Execution time: 17 ms